src/common.c

```diff
@@ -212,13 +212,13 @@ sr_ds_handle_init(struct sr_ds_handle_s **ds_handles, uint32_t *ds_handle_count)
         tmp = getenv("SR_PLUGINS_PATH");
         if (tmp) {
             if ((len = strlen(tmp)) < SR_PATH_MAX) {
-                strncpy(plugins_dir, tmp, len);
+                strncpy(plugins_dir, tmp, SR_PATH_MAX);
             } else {
                 sr_errinfo_new(&err_info, SR_ERR_INVAL_ARG, "SR_PLUGINS_PATH (%s) cannot be longer than %u.", tmp, SR_PATH_MAX);
                 goto cleanup;
             }
         } else {
-            strncpy(plugins_dir, SR_PLG_PATH, SR_PATH_MAX - 1);
+            strncpy(plugins_dir, SR_PLG_PATH, SR_PATH_MAX);
```

Comment:
The reason for adding `-1` is that if the string is `SR_PATH_MAX` chars long, it would create a non-terminated string (detected by `coverity`). It cannot occur in the first case since the string length is checked, but it can in this one, in theory.

Reply:
You are right. However, that would lead to silently truncating the value, which may lead to unpredictable behavior and may not be easy to debug. So, I have added a length check for the default value as well.